### PR TITLE
COTECH-1130 | Get size from LocalFile object if possible

### PR DESCRIPTION
**Pull request by latata**
https://fandom.atlassian.net/browse/COTECH-1130

Changes based on https://github.com/StarCitizenWiki/mediawiki-extensions-EmbedVideo

### CODE REVIEW

**Review by mszabo-wikia — APPROVED** [review]
I think we could even remove the instanceof condition, since getWidth() and co. are defined on File

**Comment by latata** [comment]
@mszabo-wikia While testing uploading the $file was an instance of `FSFile` which does getWidth etc. 🤷‍♂️

**Review by mszabo-wikia — APPROVED** [review]

**Review by nandy-andy — APPROVED** [review]
I needed a moment to understand 😅 

So, `VideoHandler::getImageSize()` used to create `FFProbe` instance every time someone called `getImageSize()`. Then called `FFProbe::getStream()` which under the hood calls `FFProbe::getMetaData()` which calls `FFProbe::invokeFFProbe()` which can be heavy and we talked about at the refinement 👌

**Review by dzawada — APPROVED** [review]

File: classes/media/FFProbe.php [hunk 1]
```diff
@@ -35,23 +42,11 @@ class FFProbe {
 	 * @param  \File MediaWiki File
```
Comment (andrzejkupczyk) [inline classes/media/FFProbe.php]: Missing `@param string $filename` tag

File: classes/media/FFProbe.php [hunk 2]
```diff
@@ -129,27 +124,58 @@ private function getFilePath() {
...
 		}
-		return true;
+
+		return [];
+	}
+
+	public function loadMetaData( string $select = 'v:0' ): bool {
+		$cache = MediaWikiServices::getInstance()->getMainWANObjectCache();
+		$cacheKey = $cache->makeGlobalKey( 'EmbedVideo', 'ffprobe', $this->filename, $select );
+		$ttl = ( $this->file instanceof \File || is_string( $this->file ) )
```
Comment (andrzejkupczyk) [inline classes/media/FFProbe.php:151]: Based on the DocBlocks for the constructor and property, `$file` should always be an instance of `File`. We need to decide whether to update the DocBlocks or to fix the code.

File: classes/media/AudioHandler.php [hunk 3]
```diff
@@ -236,4 +236,33 @@ public function getLongDesc($file) {
...
+	 * @return array
+	 */
+	protected function getFFProbeResult( $file, string $select = 'v:0' ): array {
+		$path = $file;
+
+		if ( $file instanceof \File || $file instanceof \FSFile ) {
+			$path = $file->getPath();
+		}
+
+		if ( $file === false ) {
```
Comment (andrzejkupczyk) [inline classes/media/AudioHandler.php:250]: Could you clarify when `$file` could be `false`? The DocBlock indicates that it should always be a `string`, `FSFile`, or `File`. If it doesn't meet these types, shouldn't we avoid calling this method altogether? Alternatively, we could return early in the method. Executing all the code above seems redundant if the type is incorrect.

**Review by andrzejkupczyk — APPROVED** [review]

**Review by dzawada — APPROVED** [review]

**Pull request merged**